File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int profit = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] <= min) {
                min = prices[i];
            }
            else {
                int max = prices[i] - min;
                if (max > profit) {
                    profit = max;
                }
            }
        }
        return profit;
            
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min = prices[0]
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] < min:
                min = prices[i]
            max = prices[i] - min
            if max > profit:
                profit = max
        return profit
                